### ISSUE TITLE
Add UI option to automatically set the node-name when running rancher agent on the same node as rancher server 

### ISSUE THREAD

**Issue by massimo-cannavo** [rancher/rancher#14898]
**Rancher versions:**
rancher/rancher: **v2.0.6**
rancher/agent or rancher/rancher-agent: **v2.0.6**

**Steps to Reproduce:**

1. Deploy 2 Linux hosts with the same hostname but different domain.
2. Deploy Rancher on 1 of the hosts.
3. Add cluster on that same host from step 2. (**massimo-dev**)
4. Add cluster using command generated from Rancher on 2nd host (**massimo-prod**)

**massimo-dev:**

```
# Your system has configured 'manage_etc_hosts' as True.
# As a result, if you wish for changes to this file to persist
# then you will need to either
# a.) make changes to the master file in /etc/cloud/templates/hosts.debian.tmpl
# b.) change or remove the value of 'manage_etc_hosts' in
#     /etc/cloud/cloud.cfg or cloud-config from user-data
#
127.0.1.1 massimo-server.local massimo-server
127.0.0.1 localhost

# The following lines are desirable for IPv6 capable hosts
::1 ip6-localhost ip6-loopback
fe00::0 ip6-localnet
ff00::0 ip6-mcastprefix
ff02::1 ip6-allnodes
ff02::2 ip6-allrouters
ff02::3 ip6-allhosts
```

**massimo-prod:**

```
# Your system has configured 'manage_etc_hosts' as True.
# As a result, if you wish for changes to this file to persist
# then you will need to either
# a.) make changes to the master file in /etc/cloud/templates/hosts.debian.tmpl
# b.) change or remove the value of 'manage_etc_hosts' in
#     /etc/cloud/cloud.cfg or cloud-config from user-data
#
127.0.1.1 massimo-server.net massimo-server
127.0.0.1 localhost

# The following lines are desirable for IPv6 capable hosts
::1 ip6-localhost ip6-loopback
fe00::0 ip6-localnet
ff00::0 ip6-mcastprefix
ff02::1 ip6-allnodes
ff02::2 ip6-allrouters
ff02::3 ip6-allhosts
```

**Results:**

The second host will fail to deploy **rancher/agent** because the **requestedHostname** is not a **FQDN** but rather the short version. Hence, you will have two nodes with **requestedHostname** set to massimo-server.

```
+ docker start kubelet
Error response from daemon: {"message":"No such container: kubelet"}
Error: failed to start containers: kubelet
+ sleep 2
+ docker start kubelet
Error response from daemon: {"message":"No such container: kubelet"}
Error: failed to start containers: kubelet
+ sleep 2
+ docker start kubelet
Error response from daemon: {"message":"No such container: kubelet"}
Error: failed to start containers: kubelet
+ sleep 2
+ docker start kubelet
Error response from daemon: {"message":"No such container: kubelet"}
Error: failed to start containers: kubelet
+ sleep 2
+ docker start kubelet
Error response from daemon: {"message":"No such container: kubelet"}
Error: failed to start containers: kubelet
+ sleep 2
+ docker start kubelet
Error response from daemon: {"message":"No such container: kubelet"}
Error: failed to start containers: kubelet
+ sleep 2
+ docker start kubelet
Error response from daemon: {"message":"No such container: kubelet"}
Error: failed to start containers: kubelet
+ sleep 2
+ docker start kubelet
Error response from daemon: {"message":"No such container: kubelet"}
Error: failed to start containers: kubelet
+ sleep 2
```

The issue seems to be from this line of code:

https://github.com/rancher/rancher/blob/master/package/run.sh#L117

**Note:** There is a customer that uses same hostnames but with different domains. Same naming convention in seperate environments often.

Should we use **FQDN** or stick with the short hostname?

**Comment by massimo-cannavo** [comment]
[massimo-dev.tar.gz](https://github.com/rancher/rancher/files/2251589/massimo-dev.tar.gz)
[massimo-prod.tar.gz](https://github.com/rancher/rancher/files/2251590/massimo-prod.tar.gz)

**Comment by superseb** [comment]
I don't think we can just change the way we retrieve hostnames between minor versions. Especially because `-f` is more error prone (requires proper `/etc/hosts` entry). If you want to override a hostname, you can use:

```
        -n | --node-name)        shift; CATTLE_NODE_NAME=$1         ;;
```

**Comment by massimo-cannavo** [comment]
@superseb thanks for the heads up, I will try that out. Okay so by default short hostname will be used and if the customer for some weird reason is using same hostnames but on different domains can just override it using the ```-n | --node-name flag```.

**Comment by cjellick** [comment]
@massimo-cannavo since this is available as a CLI option, im going to mark it as an enhancement and move it out of the release.

**Comment by massimo-cannavo** [comment]
@cjellick thank you and sorry for the confusion. Much appreciated!

**Comment by westlywright** [comment]
This was fixed in afe276a186c6ec59984436ce20a43d2e760d5ce9

**Comment by davidnuzik** [comment]
Version: master 2.2 (12/11/18)
Confirmed new UI feature. Validating this issue is only for the UI part of things -- since there is a way to fix this via CLI this is not covered in testing of this issue. There is a new addition (see my screenshot below and resulting node name) that lets you set the node name via the UI. This was tested and is working.
![image](https://user-images.githubusercontent.com/45179589/49822319-03e3ee80-fd3a-11e8-8a6b-79929438b963.png)
![image](https://user-images.githubusercontent.com/45179589/49822346-11997400-fd3a-11e8-971d-5172b2066a6c.png)
